website/content/docs/security/acl/acl-system.mdx

```diff
@@ -22,6 +22,10 @@ To learn how to setup the ACL system on an existing Consul datacenter, use the [
 ## ACL System Overview
 
 The ACL system is designed to be easy to use and fast to enforce while providing administrative insight.
+The diagram below shows the relationships between most of the components of the ACL system:
+
+![ACL system component relationships](/img/acl-token-policy-rule-relationship.png)
```

Comment:
Nice diagram! I think this does a great job of showing the relationship. It doesn't include Roles, or the Service/Node Identities, but I think that's probably good. Trying to include it all up front is probably confusing.

This is the diagram I made for the contrib docs that includes most thing and their relationships: https://github.com/hashicorp/consul/tree/main/contributing/acl#acl-entities

I think that level of detail is good for contributors, but probably way too much for users.

Reply:
Good artists copy, as they say. Glad this already exists on the Learn site. Just re-using! I noticed that it doesn't include roles, either because roles aren't mentioned in the Learn tutorial or because the tutorial was written before roles were a thing in Consul.

I was very tempted to add roles and service/node identities. Especially roles ... I think they are an important thing to surface. But to get 80% of the value in 2% of the time, I decided to forego that for now.

And I'm glad you created a more developer-focused diagram for contributors!

Reply:
minor suggestion: It might be better to place the image after the paragraph describing ACL Policies and Tokens so that the reader has an understanding of what these components are before viewing the image.

Reply:
@trujillo-adam: do you have a preference on whether images like this should generally be just before or just after the content (when the content is more than a few sentences long)?

I lean towards having the image first to provide an overview so that the details in the text below are easier to understand / more likely to stick, but this isn't a strong opinion.

Reply:
I could go either way. It's really a matter of which learning style do we want to appeal to. Some users appreciate having some kind of spatial representation for attaching the concepts they are about to read--even if it's not immediately clear what they mean. This specific image is actually great for the upfront approach because it uses concrete labels/parameters (rather than some of the glossier branded boxes). Other users may want to try to understand the concept syntactically and use the diagram to reinforce the understanding. The point might also be moot because I think that users are going to read the entire section more than once before it starts to coalesce. That said, I with @jkirschner-hashicorp on leaving it where it is.